db/compaction/compaction_picker.cc

```diff
@@ -721,14 +721,14 @@ Compaction* CompactionPicker::CompactRange(
   // two files overlap.
   if (input_level > 0) {
     const uint64_t limit = mutable_cf_options.max_compaction_bytes;
-    uint64_t input_level_total = 0;
     int hint_index = -1;
-    InternalKey* smallest = nullptr;
+    assert(!inputs.empty());
+    // Always include first file for progress.
+    uint64_t input_level_total = inputs[0]->fd.GetFileSize();
+    InternalKey* smallest = &(inputs[0]->smallest);
     InternalKey* largest = nullptr;
-    for (size_t i = 0; i + 1 < inputs.size(); ++i) {
-      if (!smallest) {
-        smallest = &inputs[i]->smallest;
-      }
+    for (size_t i = 1; i < inputs.size(); ++i) {
+      // Consider whether to include inputs[i]
       largest = &inputs[i]->largest;
 
       uint64_t input_file_size = inputs[i]->fd.GetFileSize();
```

Comment:
Question: Do we know that `output_level` always equals `input_level + 1`? If not, should we be calling `GetOverlappingInputsRangeBinarySearch` on all levels from `input_level + 1` up through `output_level`? Otherwise, I think we are underestimating the size of the compaction, no?

Reply:
This seems to be an implicit assumption from how CompactRangeInternal() is implemented (it's compacting from l to l+1). Added some comments and assertions for this.